aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -200,20 +201,28 @@ pub(crate) fn is_valid_txn_arg(
     module_storage: &impl AptosModuleStorage,
     ty: &Type,
     allowed_structs: &ConstructorMap,
-) -> bool {
+) -> PartialVMResult<bool> {
     use move_vm_types::loaded_data::runtime_types::Type::*;
 
-    match ty {
+    Ok(match ty {
```

Comment:
The comment of the fn says "whether argument is valid / allowed and whether it needs construction"
Is it an actual &? so true means it is valid and does not need construction? Let's clarify the comment (maybe also adding error condition)

Reply:
No, true means it is valid, e.g., an integer, or struct has a name for which we have a constructor. Updated the comment to make it more clear.